### PR TITLE
feat(orchestrator): add OpenAPI v2 implementations

### PR DESCRIPTION
**Pull request by gciavarrini**
Add v2 endpoints' implementation and unit tests.

### CODE REVIEW

File: plugins/orchestrator-common/scripts/openapi.sh
Comment (ydayagi) [inline plugins/orchestrator-common/scripts/openapi.sh:1]: I think that this file already exists in another PR

File: plugins/orchestrator-common/src/auto-generated/api/definition.ts
Comment (ydayagi) [inline plugins/orchestrator-common/src/auto-generated/api/definition.ts:1]: I think that this file already exists in another PR

File: plugins/orchestrator-common/src/auto-generated/api/models/schema.ts
Comment (ydayagi) [inline plugins/orchestrator-common/src/auto-generated/api/models/schema.ts:1]: I think that this file already exists in another PR

**Review by ydayagi — APPROVED** [review]

File: plugins/orchestrator-common/scripts/openapi.sh
Comment (gciavarrini) [reply plugins/orchestrator-common/scripts/openapi.sh:1]: This is a draft PR. It depends on PR #1123, as per description. This means that it is built on top of it.
Once PR #1123 is merged, i can rebase this PR on top of main, so you won't see this change.

File: plugins/orchestrator-common/src/auto-generated/api/definition.ts
Comment (gciavarrini) [reply plugins/orchestrator-common/src/auto-generated/api/definition.ts:1]: This is a draft PR. It depends on PR #1123, as per description. This means that it is built on top of it.
Once PR #1123 is merged, i can rebase this PR on top of main, so you won't see this change.

File: plugins/orchestrator-backend/dist-dynamic/yarn.lock
Comment (ydayagi) [inline plugins/orchestrator-backend/dist-dynamic/yarn.lock:1]: i think we should not push this file

**Review by ydayagi — CHANGES_REQUESTED** [review]

File: plugins/orchestrator-backend/dist-dynamic/yarn.lock
Comment (gciavarrini) [reply plugins/orchestrator-backend/dist-dynamic/yarn.lock:1]: It must be pushed IMO.
Every time you change the package.json from the backend plugin, you need to regenerate the package.json and yarn.lock files inside the dist-dynamic folder

Comment (batzionb) [reply plugins/orchestrator-backend/dist-dynamic/yarn.lock:1]: @ydayagi why do you think this change should not be pushed?
What's the difference between this case and any other nodejs application?
The yarn.lock needs to be updated to insure that when the application is installed it gets the correct versions of all the node_modules tree

**Comment by caponetto** [comment]
@batzionb is taking care of the review.

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      return 'Error';
+    case ProcessInstanceState.Completed.valueOf():
+      return 'Completed';
+    case ProcessInstanceState.Aborted.valueOf():
+      return 'Aborted';
+    case ProcessInstanceState.Suspended.valueOf():
+      return 'Suspended';
+    default:
+      throw new Error(
+        'state is not one of the vlalues of type ProcessInstanceStatusDTO',
```
Comment (batzionb) [inline plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts:132]:
```suggestion
        'state is not one of the values of type ProcessInstanceStatusDTO',
```

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  WorkflowSpecFile,
+  WorkflowSpecFileDTO,
+} from '@janus-idp/backstage-plugin-orchestrator-common';
+
+// Mapping functions
+export function mapToWorkflowOverviewDTO(
+  overview: WorkflowOverview,
+): WorkflowOverviewDTO {
+  return {
+    workflowId: overview.workflowId,
```
Comment (batzionb) [inline plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts:29]: you can use object spread to make this shorter
return {
  ...overview,
  category: mapWorkflowCategoryDTOFromString(overview.category),
}
same in other mapping functions

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    category: mapWorkflowCategoryDTOFromString(overview.category),
+    avgDurationMs: overview.avgDurationMs,
+    description: overview.description,
+  };
+}
+
+export function mapWorkflowCategoryDTOFromString(
+  category?: string,
+): WorkflowCategoryDTO {
+  const lowerCase = category?.toLocaleLowerCase();
```
Comment (batzionb) [inline plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts:43]: how about 
return category?.toLocalLowerCase === "assessment" ? "assessment" : "infrastructure";

Comment (gciavarrini) [reply plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts]: done

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts (hunk 1 above)
Comment (gciavarrini) [reply plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts]: thanks! fixed

File: plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts (hunk 2 above)
Comment (gciavarrini) [reply plugins/orchestrator-backend/src/service/api/mapping/V2Mappings.ts]: I agree that it looks nicer, but if it's ok for you, I'd prefer to do these changes in a following PR to be sure I won't break anything (I'll add more unit tests).

**Review by batzionb — APPROVED** [review]

**Pull request merged**